src/main/java/gregtech/client/ClientProxy.java

```diff
@@ -180,58 +185,33 @@ public static void addMaterialFormulaHandler(ItemTooltipEvent event) {
             MetaOreDictItem oreDictItem = (MetaOreDictItem) itemStack.getItem();
             Optional<String> oreDictName = OreDictUnifier.getOreDictionaryNames(itemStack).stream().findFirst();
             if (oreDictName.isPresent() && oreDictItem.OREDICT_TO_FORMULA.containsKey(oreDictName.get())) {
-                chemicalFormula = oreDictItem.OREDICT_TO_FORMULA.get(oreDictName.get());
+                tooltips.add(TextFormatting.YELLOW + oreDictItem.OREDICT_TO_FORMULA.get(oreDictName.get()));
             }
         } else if (unificationEntry != null && unificationEntry.material != null) {
-            chemicalFormula = unificationEntry.material.getChemicalFormula();
+            tooltips.add(TextFormatting.YELLOW + unificationEntry.material.getChemicalFormula());
         } else if (ItemNBTUtils.hasTag(itemStack)) { // Test for Fluids
             // Vanilla bucket
-            fluidTooltips = FluidTooltipUtil.getFluidTooltip(ItemNBTUtils.getString(itemStack, "FluidName"));
-
-            if(fluidTooltips != null) {
-                chemicalFormula = fluidTooltips.get(0);
-                temperature = fluidTooltips.get(1);
-                isGas = fluidTooltips.get(2);
-            }
+            tooltips = FluidTooltipUtil.getFluidTooltip(ItemNBTUtils.getString(itemStack, "FluidName"));
 
             // GTCE Cells, Forestry cans, some other containers
-            if (chemicalFormula == null) {
+            if (tooltips != null && tooltips.size() > 2 && tooltips.get(2) == null) {
```

Comment:
This final check for `tooltips.get(2) == null` is incorrect I believe. It prevents tooltips from showing up on our cells for most fluids, as out fluids have valid states.